### ISSUE TITLE
Cannot update stripe to 21.8.0

### ISSUE THREAD

**Issue by xsalefter** [killbill/killbill-stripe-plugin#87]
See [this PR](https://github.com/killbill/killbill-stripe-plugin/pull/86). Exception from log:

```
Error:  org.killbill.billing.plugin.stripe.TestStripePaymentPluginApi.testVerifySyncOfPaymentMethods  Time elapsed: 1.222 s  <<< FAILURE!
com.stripe.exception.ApiKeyMissingException: API key is not set for autoPagingIterable. You can set the API key globally using Stripe.ApiKey, or through RequestOptions with autoPagingIterable(params, options).
	at com.stripe.model.StripeCollection.autoPagingIterable(StripeCollection.java:63)
	at org.killbill.billing.plugin.stripe.StripePaymentPluginApi.getPaymentMethods(StripePaymentPluginApi.java:489)
```

This maybe have similarity with this issue: https://github.com/stripe/stripe-java/issues/1176

**Comment by xsalefter** [comment]
So, yes, original stripe issue above have something to do with this: that `stripe-java`, starting [v21](https://github.com/stripe/stripe-java/wiki/Migration-guide-for-v21), requiring `ApiKey`, either set globally (via `Stripe.apiKey`) or set it in `RequestOptions` and pass it in **every** `StripeCollection.autoPagingIterable` and `StripeSearchResult.autoPagingIterable`

In our `StripePaymentPluginApi`, we can modify so that [this line](https://github.com/killbill/killbill-stripe-plugin/blob/work-for-release-0.23.x/src/main/java/org/killbill/billing/plugin/stripe/StripePaymentPluginApi.java#L489) will looks like like this:
```java
final Iterable<PaymentMethod> stripePaymentMethodsCard = PaymentMethod
  .list(paymentMethodParams, requestOptions)
  .autoPagingIterable(paymentMethodParams, requestOptions);
```

But unfortunately, we have `StripePluginProperties` that [also use](https://github.com/killbill/killbill-stripe-plugin/blob/work-for-release-0.23.x/src/main/java/org/killbill/billing/plugin/stripe/StripePluginProperties.java#L256) `#autoPagingIterable()`, and it's not so trivial adding `requestOptions` there, since it uses by `StripeDao`. 

Solutions that I can think:

1. Change `StripePluginProperties#toAdditionalDataMap()` and add additional method parameter to set `lastCharge` properties. Add this logic in every methods that call `StripeDao.addResponse()` and `StripeDao.updateResponse()`.

2. Set Stripe ApiKey globally in `StripeConfigProperties` [constructor](https://github.com/killbill/killbill-stripe-plugin/blob/work-for-release-0.23.x/src/main/java/org/killbill/billing/plugin/stripe/StripeConfigProperties.java#L78).

@pierre WDYT?

**Comment by pierre** [comment]
Good 🕵️  work!

I don't think we can set it globally: the plugin needs to be multi-tenant, and different tenants could use different api keys.

Do we really need to use `stripePaymentIntent.getCharges().autoPagingIterable()` to find the last charge? Is there maybe a smarter way of doing this?

If not, 1. seems the only option. It would actually be nice architecturally, that way `StripePluginProperties` doesn't do network calls, it's just a data transformation layer. I'm not sure the ramifications though and how hard it is to pass in `lastCharge` (e.g. if lots of code changes?)...

**Comment by xsalefter** [comment]
> Do we really need to use stripePaymentIntent.getCharges().autoPagingIterable() to find the last charge?

From [stripe docs](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-charges-data), get charge collection from `PaymentIntent`  is efficient enough to get `Charge` collection:

`This list only contains the latest charge, even if there were previously multiple unsuccessful charges. `

We can use `Charge` object instead:

```java
Charge.list(ChargeListParams.builder().setPaymentIntent(stripePaymentIntent.getId()).setLimit(1L).build());
```

But we'll need extra care to filter unsuccessful charge, and any other parameter that we might missing.

**Comment by xsalefter** [comment]
I think this issue already solved by these commits: e051753e to 02cac689
